apps/test/unit/javalab/JavalabEditorTest.js

```diff
@@ -169,33 +171,107 @@ describe('Java Lab Editor Test', () => {
       const javalabEditor = editor.find('JavalabEditor').instance();
       javalabEditor.setState({
         activeTabKey: 'file-0',
-        tabs: [
-          {
-            filename: 'file0.java',
-            key: 'file-0'
-          },
-          {
-            filename: 'file1.java',
-            key: 'file-1'
-          }
-        ],
+        orderedTabKeys: ['file-0', 'file-1'],
+        fileMetadata: {
+          'file-0': 'file1.java',
+          'file-1': 'file2.java'
+        },
         showMenu: true,
         contextTarget: 'file-0'
       });
       javalabEditor.onOpenFile('file-1');
       expect(javalabEditor.state.activeTabKey).to.equal('file-1');
-      expect(javalabEditor.state.tabs).to.deep.equal([
-        {
-          filename: 'file1.java',
-          key: 'file-1'
-        },
-        {
-          filename: 'file0.java',
-          key: 'file-0'
-        }
+      expect(javalabEditor.state.orderedTabKeys).to.deep.equal([
+        'file-1',
+        'file-0'
       ]);
       expect(javalabEditor.state.showMenu).to.be.false;
       expect(javalabEditor.state.contextTarget).to.be.null;
     });
   });
+
+  describe('Create New File', () => {
+    it('updates state and sources on create save', () => {
```

Comment:
Perhaps also add a check that we can't create a file with a name that already exists.